tests/kernel/sched/preempt/src/main.c

```diff
@@ -102,8 +102,8 @@ void wakeup_src_thread(int id)
 	for (int i = 0; i < NUM_THREADS; i++) {
 		k_tid_t th = &worker_threads[i];
 
-		zassert_true(th->base.thread_state & _THREAD_PENDING,
-			     "worker thread %d not pending?", i);
+		zassert_equal(k_thread_state_str(th), "pending",
+				"worker thread %d not pending?", i);
```

Comment:
This has to be a strcmp().  Right now this passes only because the compiler optimizes the string literals to the same pointer value.  That optimization can be trivially defeated as the code evolves (e.g. imagine a future improvement that puts the kernel strings into a separate flash segment or something that doesn't need to be updated along with the code).

Reply:
I agree with Andy. Please take look at zassert_mem_equal(), which is what you want I think. If it does not work for you, use strcmp() as Andy suggested.

Reply:
@andyross I think that assert should say to the programmer that here is an error, without any assumptions. Why necessary to use a question mark in that sentence "worker thread %d not pending?"

Reply:
> This has to be a strcmp(). 

@andyross  I modified it using strcmp() function, please take a review of it.

Reply:
LGTM.